test/unit/io_deviceManager.js

```diff
@@ -19,3 +19,16 @@ test('default connected', t => {
     t.strictEqual(deviceManager.isConnected, true);
     t.end();
 });
+
+test('list', t => {
+    const deviceManager = new DeviceManager();
+    deviceManager
+        .list('test', 'test', null)
+        .catch(err => {
+            t.true((typeof err === 'undefined') || (typeof err === 'object'));
+        })
+        .then(body => {
+            t.true((typeof body === 'undefined') || (typeof body === 'object'));
+            t.end();
+        });
+});
```

Comment:
Is the idea with this test that the `list(...)` will fail (either because the SDM isn't running or because it doesn't offer any `test` plugin) so you'll end up in the `catch` clause and just verify the "shape" of the error?

If so: I think it's a little deceptive that the expected path involves running through both the `catch`, which "returns" `undefined`, and the `then`, which receives that `undefined`. I'd recommend using the two-clause form of `then`... maybe something like this:
```js
deviceManager
    .list('test', 'test', null);
    .then(
        body => {
            // body should always be an object -- never undefined or null
            // or maybe we just always fail in this case since success is unexpected?
            t.end();
        },
        err => {
            t.true(...); // failure is expected and should pass the test
            t.end();
        }
    );
```